File: DesignPatternsExercises/ruby/abstract_factory_design_pattern/exercise/phone_number_factory.rb
require_relative 'us_phone_number'
require_relative 'uk_phone_number'

class PhoneNumberFactory
  def __________(6)__________(country, phone_number)
    return __________(7)__________(phone_number) if country == :US
    return __________(8)__________(phone_number) if country == :UK

    raise 'Country is not supported'
  end
end

File: DesignPatternsExercises/java/AbstractFactoryDesignPattern/Solution/PhoneNumberFactory.java
public class PhoneNumberFactory implements AbstractFactory {
    @Override
    public PhoneNumber getPhoneNumberInstance(Country country, String phoneNumber) {
        if (country == Country.US)
            // return __________(7)__________(phoneNumber);
            return new USPhoneNumber(phoneNumber);
        if (country == Country.UK)
            // return __________(8)__________(phoneNumber);
            return new UKPhoneNumber(phoneNumber);
        return null;
    }

    @Override
    public PostCode getPostCodeInstance(Country country, String postalCode) {
        throw new UnsupportedOperationException("Unable to get PostCodeInstance in the PhoneNumberFactory");
    }
}
